inc/saiipsec.h

```diff
@@ -0,0 +1,1047 @@
+} sai_ipsec_port_stat_t;
+
+/**
+ * @brief Attribute Id for sai_ipsec_sa
+ */
+typedef enum _sai_ipsec_sa_attr_t
+{
+    /**
+     * @brief Start of IPsec Security Association attributes
+     */
+    SAI_IPSEC_SA_ATTR_START,
+
+    /**
+     * @brief IPsec direction
+     *
+     * @type sai_ipsec_direction_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_IPSEC_SA_ATTR_IPSEC_DIRECTION = SAI_IPSEC_SA_ATTR_START,
+
+    /**
+     * @brief IPsec object id
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_IPSEC
+     */
+    SAI_IPSEC_SA_ATTR_IPSEC_ID,
+
+    /**
+     * @brief SA status.
+     *
+     * @type sai_ipsec_sa_status_t
+     * @flags READ_ONLY
+     */
+    SAI_IPSEC_SA_ATTR_STATUS,
+
+    /**
+     * @brief Externally assigned SA Index value for this Security Association.
+     * Used only when SAI_IPSEC_ATTR_EXTERNAL_SA_INDEX_ENABLE == true.
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_AND_SET
+     * @default 0
+     */
+    SAI_IPSEC_SA_ATTR_EXTERNAL_SA_INDEX,
+
+    /**
+     * @brief SA Index value for this Security Association.
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_IPSEC_SA_ATTR_SA_INDEX,
+
+    /**
+     * @brief List of IPsec ports for this SA.
+     *
+     * @type sai_object_list_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_IPSEC_PORT
+     * @default empty
+     */
+    SAI_IPSEC_SA_ATTR_IPSEC_PORT_LIST,
+
+    /**
+     * @brief SPI value for this Security Association, carried in ESP header.
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_IPSEC_SA_ATTR_IPSEC_SPI,
+
+    /**
+     * @brief Enable 64-bit ESN (vs 32-bit SN) for this Security Association
+     *
+     * @type bool
+     * @flags CREATE_ONLY
+     * @default true
+     */
+    SAI_IPSEC_SA_ATTR_IPSEC_ESN_ENABLE,
+
+    /**
+     * @brief Cipher suite for this SA.
+     *
+     * @type sai_ipsec_cipher_t
+     * @flags CREATE_ONLY
+     * @default SAI_IPSEC_CIPHER_AES256_GCM16
+     */
+    SAI_IPSEC_SA_ATTR_IPSEC_CIPHER,
+
+    /**
+     * @brief IPsec Traffic Encryption Key used for encryption/decryption.
+     * Network Byte order. AES128 uses only Bytes 16..31.
+     *
+     * @type sai_macsec_sak_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_IPSEC_SA_ATTR_MACSEC_SAK,
+
+    /** @ignore - proper name for IPsec encryption key */
+    SAI_IPSEC_SA_ATTR_ENCRYPTION_KEY = SAI_IPSEC_SA_ATTR_MACSEC_SAK,
+
+    /**
+     * @brief IPsec Salt portion of IV
+     * Network Byte order.
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_IPSEC_SA_ATTR_SALT,
+
+    /**
+     * @brief IPsec Authentication Key
+     * Network Byte order.
+     *
+     * @type sai_macsec_auth_key_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_IPSEC_SA_ATTR_MACSEC_AUTH_KEY,
```

Comment:
Similar to earlier comment, why is MACSEC used in the key name?

Reply:
Same reason as in case of SAI_IPSEC_SA_ATTR_MACSEC_SAK.

Reply:
Response is similar as that for MACSEC_SAK

Reply:
Sam response as comment for SAI_IPSEC_SA_ATTR_MACSEC_SAK.